x-pack/plugins/discover_log_explorer/common/entity_list.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+type Predicate<EntityType> = (item: EntityType) => boolean;
+type SortDirection = 'asc' | 'desc';
+
+export class EntityList<EntityType extends Record<string, any>> {
```

Comment:
Does this make the code that handles such lists signidicantly cleaner? Or does it even hide details that the developer should better know (such as modifying the list given in the constructor in place).

What do you think about just creating a pure function that creates a comparator instead?

Reply:
I'm not sure I got how you'd like to have this implemented, do you mean simplifying the code using a function that filters and sorts instead of using this class?

Reply:
Sorry for giving little context. So here's the single usage that I could find:

```typescript
new EntityList<Dataset>(integration.datasets)
  .filterBy((stream) => Boolean(stream.title?.includes(nameQuery ?? '')))
  .sortBy('name', sortOrder)
  .build()
```

The reader needs to understand what `EntityList` and its methods do. Compare to

```typescript
integration.datasets
  .filter((stream) => Boolean(stream.title?.includes(nameQuery ?? '')))
  .sort(compareByField('name', sortOrder))
```

which allows the reader to use their pre-existing JS knowledge to fully understand what these lines do.

Reply:
It's not necessarily about mutating it, but about hiding that fact from the developer. The replacement I proposed also mutates in place, but that's not surprising to anyone familiar with JS.

Reply:
I see your point, I'm good with apply this suggestion as this is the only use I have for this utility 👌